internal/pkg/composable/providers/kubernetessecrets/kubernetes_secrets.go

```diff
@@ -72,13 +121,20 @@ func (p *contextProviderK8sSecrets) Fetch(key string) (string, bool) {
 		return "", false
 	}
 	ns := tokens[1]
+	reader, err := p.k8sReaderProvider.getReader(ns)
+	if err != nil {
+		p.logger.Errorf("Could not create K8S client: %v", err)
+		return "", false
+	}
 	secretName := tokens[2]
 	secretVar := tokens[3]
 
-	secretIntefrace := client.CoreV1().Secrets(ns)
-	ctx := context.TODO()
-	secret, err := secretIntefrace.Get(ctx, secretName, metav1.GetOptions{})
-	if err != nil {
+	secret := corev1.Secret{}
+	if err := reader.Get(context.TODO(), client.ObjectKey{Namespace: ns, Name: secretName}, &secret); err != nil {
+		if k8serrors.IsNotFound(err) {
+			p.logger.Errorf("Secret %s/%s not found: %v", ns, secretName, err)
+			return "", false
+		}
 		p.logger.Errorf("Could not retrieve secret from k8s API: %v", err)
```

Comment:
Could be Debug level as this might create too much noise and false positives

Reply:
I applied your suggestion. I'm not really familiar with how we want to surface such situations, this was to be consistent with the current behaviour:

```diff
-       secretIntefrace := client.CoreV1().Secrets(ns)
-       ctx := context.TODO()
-       secret, err := secretIntefrace.Get(ctx, secretName, metav1.GetOptions{})
-       if err != nil {
-               p.logger.Errorf("Could not retrieve secret from k8s API: %v", err)
+       secret := corev1.Secret{}
+       if err := reader.Get(context.TODO(), client.ObjectKey{Namespace: ns, Name: secretName}, &secret); err != nil {
+               if k8serrors.IsNotFound(err) {
+                       p.logger.Debugf("Secret %s/%s not found: %v", ns, secretName, err)
+                       return "", false
+               }
+               p.logger.Debugf("Could not retrieve secret from k8s API: %v", err)
```